docs/cstor.md

```diff
@@ -0,0 +1,384 @@
+---
+id: cstor
+title: cStor Overview
+sidebar_label: cStor
+---
+
+------
+
+cStor is the recommended storage engine in OpenEBS. cStor was originally introduced in OpenEBS 0.7 release and has been tested well in the community and in production deployments. The primary function of cStor is to serve the iSCSI block storage using the underlying disks in a cloud native way. cStor is a very light weight and feature rich storage engine. It provides enterprise grade features such as synchrous data replication, snapshots, clones, thin provisioning of data, high resiliency of data, data consistency and on-demand increase of capacity or performance. 
+
+When the stateful application desires the storage to provide high availability of data, cStor is configured to have 3 replicas where data is written synchronously to all the three replicas. As shown below, the cStor target is replicating the data to Node1 (R1), Node3(R2) and Node4(R3). The data is written to the three repicas before the response is sent back to the application. 
+
+<br><img src="/docs/assets/cstor-for-deployment.png" alt="cStor components" width="900"/>
+
+<br>
+
+
+
+When the stateful application itself is taking care of data replication, it is typically deployed as a Kubernetes *statefulset*. For a statefulset, it is typical to configure cStor to have one replica.
+
+<br><img src="/docs/assets/cstor-for-statefulset.png" alt="cStor components" width="900"/>
+
+<br>
+
+
+
+cStor has two main components:
+
+- **cStor targets** - cStor-target resides in this pod. It primarily does two things : exposethe iSCSI LUN and handle the synchronous replication of data if it has more than one replica. There will be one cStor target pod for each volume.
+- **cStor  pools**  - cStor replicas reside in this pod. Lots of volume replicas typically share a cStor pool. A cStor pool is a set of cStor pool pods with each pod on a given node. 
```

Comment:
The statement  `A cStor pool is a set of cStor pool pods with each pod on a given node.` is not clear. 

Just thinking of rewording the section a bit as follows:

```
cStor has two main components:

(a) cStor Pool Pods: cStor Pool Pods are responsible for persisting the data on to the disks. The cStor Pool Pods are instantiated on a node and are provided with one or more disks on which data will be saved. Each cStor Pool Pod - can save data of one or more cStor Volumes. cStor Pool Pod - carves out space for each volume replica, manages the snapshots and clones of the replica. A set of such cStor Pool Pods (instances) form a single Storage Pool. The administrator will have to created a Storage Pool of type cStor - before creating a StorageClass for cStor Volumes. 

(b) cStor Target Pods:  When a cStor Volume is provisioned, it creates a new cStor Target Pod that is responsible for exposing the iSCSI LUN. cSTor Target Pod receives the data from the workloads, and then passes it on to the respective cStor Volume Replicas (on cStor Pools). cStor Target Pod handles the synchronous replication and quorum management of its replicas. 
```